Models/User.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * User class file
+ *
+ * PHP Version 7.2
+ *
+ * @category Models
+ * @package  Models
+ * @author   Hervé Boulangué <h.boulangue@gmail.com>
+ * @license  http://opensource.org/licenses/gpl-license.php GNU Public License
+ * @link     http://blog.local/
+ */
+
+namespace Models;
+
+/**
+ * Model for user
+ *
+ * @category Models
+ * @package  Models
+ * @author   Hervé Boulangué <h.boulangue@gmail.com>
+ * @license  http://opensource.org/licenses/gpl-license.php GNU Public License
+ * @link     http://blog.local/
+ *
+ * @Entity @Table(name="User")
+ */
+class User
+{
+    /**
+     * User's id into User table
+     *
+     * @Id @Column(type="integer") @GeneratedValue
+     */
+    protected $id;
+
+    /**
+     * The name of the user
+     *
+     * @var string $username the name of the user
+     *
+     * @Column(type="string")
+     */
+    private $username;
+
+    /**
+     * The hash of the password
+     *
+     * @var string $passwordHash the hash of the password
+     *
+     * @Column(type="string")
+     */
+    private static $passwordHash = '$2y$10$4F0MRbwFJsntVtHmuf3luOVH5Tj05KBqgDUSqSAc3UYI.OHuN.rj6';
+
+    /**
+     * Instantiates a new user
+     *
+     * @param string $username     the name of the user to set
+     * @param string $passwordHash the hash of the password to set
+     */
+    public function __construct(string $username, string $passwordHash)
+    {
+        $this->setUsername($username);
+        $this->setPasswordHash($passwordHash);
+    }
+
+    /**
+     * Returns a user with the given username and password or null if it doesn't exist
+     *
+     * @param string $username the name of the user to return
+     * @param string $password the password of the user to return
+     *
+     * @return User|null a user with the given username and password or null if it doesn't exist
+     */
+    public static function getUser(string $username, string $password): ?User
+    {
+        if ($username == "h.boulangue" && $password == "test") {
```

Comment:
```
$userRepository = Database::getEntityManager()->getRepository("Models\\User");
$user = $userRepository->findOneBy(
    array(
        "username" => $username
    )
);
if($user != null){
    if(password_verify($password, $user->getPasswordHash())){
        /**
         * @noinspection PhpIncompatibleReturnTypeInspection
         * Safe because $user has been generated by Doctrine
         */
        return $user;
    }
}
return null;
```